addons/mail/models/ir_attachment.py

```diff
@@ -11,6 +11,13 @@
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
+    thumbnail = fields.Image()
+    has_thumbnail = fields.Boolean(compute="_compute_has_thumbnail")
+
+    def _compute_has_thumbnail(self):
+        for a in self:
+            a.has_thumbnail = bool(a.thumbnail)
```

Comment:
Note: this will actually require reading the thumbnail on disk to determine it's existence, which can be problematic when storage is moved to a network place.

I believe adding `bin_size` to context before reading the `thumbnail` the would fix this problem (internal implementation of binary field).